src/main/java/com/google/impactdashboard/database_manager/data_read/DataReadManagerImpl.java

```diff
@@ -86,27 +97,46 @@ public double getAverageIAMBindingsInPastYear(String projectId) {
    */
   @Override
   public Map<Long, Integer> getMapOfDatesToIAMBindings(String projectId) {
+    QueryJobConfiguration queryConfiguration = queryConfigurationBuilder
+      .getDatesToBindingsConfiguration()
+      .addNamedParameter("projectId", QueryParameterValue.string(projectId))
+      .build();
+    TableResult results = database.readDatabase(queryConfiguration);
+
     HashMap<Long, Integer> datesToBindings = new HashMap<Long, Integer>();
-    AtomicReference<Long> date = new AtomicReference<Long>(1590883200000L);
+    for (FieldValueList row : results.iterateAll()) {
+      long timestamp = row.get(IAMBindingsSchema.TIMESTAMP_COLUMN)
+        .getTimestampValue() / 1000;
+      int iamBindings = (int) row.get(IAMBindingsSchema.NUMBER_BINDINGS_COLUMN).getLongValue();
+
+      datesToBindings.put(timestamp, iamBindings);
+    } 
+    return datesToBindings;
+  }
 
-    if (projectId.equals("project-id-1")) {
-      Arrays.asList(1000, 1000, 1000, 2000, 2050, 2150, 2150, 2150, 2150, 
-        1150, 1150, 1150, 1150, 2000, 2000, 2500, 2500, 2300, 2300, 1000, 
-        1000, 1000, 1100, 1100, 1000, 1000, 1300, 1300, 1350, 1350)
-        .forEach(numberBindings -> {
-          datesToBindings.put(date.get(), numberBindings);
-          date.set(date.get() + 86400000);
-        });
-    } else if (projectId.equals("project-id-2")) {
-      Arrays.asList(500, 500, 750, 750, 750, 750, 750, 1000, 1000, 
-        1000, 500, 500, 500, 600, 600, 600, 600, 300, 300, 1000, 
-        1000, 1000, 1100, 1100, 1000, 1000, 500, 500, 500, 500)
-        .forEach(numberBindings -> {
-          datesToBindings.put(date.get(), numberBindings);
-          date.set(date.get() + 86400000);
-        });
+  /**
+   * Queries the IAM database for information about the project with id 
+   * {@code projectId}, and returns a {@code ProjectIdentification} object 
+   * containing that information. 
+   */
+  private ProjectIdentification getProjectIdentificationForProject(String projectId) {
+    QueryJobConfiguration queryConfiguration = queryConfigurationBuilder
+      .getProjectIdentificationInformationConfiguration()
+      .addNamedParameter("projectId", QueryParameterValue.string(projectId))
+      .build();
+    TableResult results = database.readDatabase(queryConfiguration);
+    
+    String projectName = null;
+    String projectNumber = null;
+    for (FieldValueList row : results.iterateAll()) {
+      projectName = row.get(IAMBindingsSchema.PROJECT_NAME_COLUMN).getStringValue();
+      projectNumber = row.get(IAMBindingsSchema.PROJECT_NUMBER_COLUMN).getStringValue();
     }
 
```

Comment:
Are you expecting only one result? 

If not, this method would seem to only take the last projectName and projectNumber, which doesn't quite seem right?

Reply:
I'm expecting only one result---the SQL query searches for a given project id and set the query to limit 1, so I just read off the project name and number of that single row.